app/services/gladly/api/client.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'net/http'
+
+module Gladly
+  module Api
+    class Client
+      include Gladly::Api::ErrorHandling
+
+      def initialize(payload: {})
+        @api_username = SpreeGladly::Config.gladly_api_username
+        @api_key = SpreeGladly::Config.gladly_api_key
+        @base_url = SpreeGladly::Config.gladly_api_base_url
+        @payload = payload
+      end
+
+      def call
+        return if base_url.blank?
+
+        perform_request
+      end
+
+      def perform_request
+        post_request if request_method.eql?(:post)
+      end
+
+      def post_request
+        uri = URI(request_url)
+
+        http = Net::HTTP.new(uri.host, uri.port)
```

Comment:
As discussed during the premortem, it would be great to also apply read timeout of 20 seconds when making requests